SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestShutdownPreparation/SpatialTestShutdownPreparationTrigger.h

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "Http.h"
+#include "SpatialFunctionalTest.h"
+#include "TestPrepareShutdownListener.h"
+#include "SpatialTestShutdownPreparationTrigger.generated.h"
+
+UCLASS()
+class SPATIALGDKFUNCTIONALTESTS_API ASpatialTestShutdownPreparationTrigger : public ASpatialFunctionalTest
+{
+	GENERATED_BODY()
+
+	ASpatialTestShutdownPreparationTrigger();
+
+	virtual void BeginPlay() override;
+
+	UPROPERTY(EditInstanceOnly, Category = "Test Settings")
+	TSubclassOf<ATestPrepareShutdownListener> PrepareShutdownListenerClass;
+
+	UPROPERTY(EditInstanceOnly, Category = "Test Settings")
+	float EventWaitTime;
```

Comment:
I could not find where this variable is set.

Reply:
Good catch - since it's a uproperty I set the actual value on the test instance in the editor, but I've now given it a default value in c++ as well, since a wait time of 0 isn't really reasonable.